tests/integration/long/test_failure_types.py

```diff
@@ -0,0 +1,295 @@
+            """
+            CREATE KEYSPACE testksfail
+            WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '3'}
+            """, consistency_level=ConsistencyLevel.ALL, expected_exception=None)
+
+        # create table
+        self._perform_cql_statement(
+            """
+            CREATE TABLE testksfail.test (
+                k int PRIMARY KEY,
+                v int )
+            """, consistency_level=ConsistencyLevel.ALL, expected_exception=None)
+
+        # Simple insert so we have something to query
+        self._perform_cql_statement(
+            """
+            INSERT INTO testksfail.test (k, v) VALUES  (1, 0 )
+            """, consistency_level=ConsistencyLevel.ALL, expected_exception=None)
+
+        # Disable one node
+        failing_nodes = [self.node1]
+        self.setFailingNodes(failing_nodes, "testksfail")
+
+        # With one node disabled we would expect a write failure with ConsistencyLevel of all
+        self._perform_cql_statement(
+            """
+            INSERT INTO testksfail.test (k, v) VALUES  (1, 0 )
+            """, consistency_level=ConsistencyLevel.ALL, expected_exception=WriteFailure)
+
+        # We have two nodes left so a write with consistency level of QUORUM should complete as expected
+        self._perform_cql_statement(
+            """
+            INSERT INTO testksfail.test (k, v) VALUES  (1, 0 )
+            """, consistency_level=ConsistencyLevel.QUORUM, expected_exception=None)
+
+        failing_nodes = []
+
+        # Restart the nodes to fully functional again
+        self.setFailingNodes(failing_nodes, "testksfail")
+
+        # Drop temporary keyspace
+        self._perform_cql_statement(
+            """
+            DROP KEYSPACE testksfail
+            """, consistency_level=ConsistencyLevel.ANY, expected_exception=None)
+
+    def test_tombstone_overflow_read_failure(self):
+        """
+        Test to validate that a ReadFailure is returned from the coordinator when a specified threshold of tombstombs is
+        reached.
+
+        test_tombstomb_overflow_read_failure First sets the tombstone failure threshold down to a level that allows it
+        to be more easily encountered. We then create some wide rows and ensure they are deleted appropriately. This
+        produces the correct amount of tombstombs. Upon making a simple query we expect to get a read failure back
+        from the coordinator.
+
+
+        @since 2.6.0
+        @jira_ticket PYTHON-238
+        @expected_result Appropriate write failures from the coordinator
+
+        @test_category queries:basic
+        """
+
+        # Setup table for "wide row"
+        self._perform_cql_statement(
+            """
+            CREATE TABLE test3rf.test2 (
+                k int,
+                v0 int,
+                v1 int, PRIMARY KEY (k,v0))
+            """, consistency_level=ConsistencyLevel.ALL, expected_exception=None)
+
+        for i in range(3000):
```

Comment:
If you care, this could be done quite a bit faster with concurrent.execute_concurrent_with_args